### ISSUE TITLE
Improve error messages for missing indicators

### ISSUE THREAD

**Issue by 0xazure** [apidoc/apidoc#159]
Hi all,

First, some background.  Previously, I was using [gulp-apidoc](https://github.com/ayhankuru/gulp-apidoc) (pinned to use [apidoc@0.6.5](https://github.com/ayhankuru/gulp-apidoc/blob/02f48423bd2d6719a12b3cea3c9f0b586936ce30/package.json#L27)) briefly before migrating to use apiDoc directly in my `gulpfile`.  I anticipated some things breaking because of this upgrade, especially because of all the changes listed in the CHANGELOG.md between version 0.6.x and 0.8.2.  However, I ran into some rather confusing output after upgrading.  I was seeing the following:

``` sh
error:   TypeError: Cannot read property 'length' of null
[21:05:10] 'doc:api' errored after 52 ms
[21:05:10] TypeError: Cannot read property 'length' of null
    at .../node_modules/apidoc/lib/workers/api_name.js:30:48
    at Array.forEach (native)
    at .../node_modules/apidoc/lib/workers/api_name.js:15:20
    at Array.forEach (native)
    at Object.postProcess (.../node_modules/apidoc/lib/workers/api_name.js:14:17)
    at .../node_modules/apidoc/lib/worker.js:95:20
    at forOwn (.../node_modules/lodash/dist/lodash.js:2105:15)
    at Function.forEach (.../node_modules/lodash/dist/lodash.js:3302:9)
    at Worker.process (.../node_modules/apidoc/lib/worker.js:92:7)
    at parse (.../node_modules/apidoc/lib/apidoc.js:182:20)
```

I initially thought it was an issue with my gulp task, but it turns out that I was missing an [`@apiName`](http://apidocjs.com/#param-api-name) indicator instead, and this was causing the parse error.  However, the fact that I was missing this indicator is not obvious from the above stacktrace; I only figured it out through some trial-and-error and some re-reading of the docs.

**TL;DR**: In my opinion, the indication that `@apiName` is required is not obvious enough in the docs.  There should also be a human-friendly error output to STDERR if a **required** indicator is missing.

Also, there appears to be some overlap between `@apiName` and the optional `title` field in `@api`, where both of these fields appear to be used for navigation and article headers (at least, according to the docs.  I haven't looked deeply at the code to see what's actually being used as of yet).  I feel as though `@apiName` is redundant, and should be merged with `@api`, which would then take a required `name` field instead of an optional `title` field.  The current use for the `title` field of "A short title", per the docs, appears to be better suited to the `@apiDescription` indicator.

Cheers.
